main.js

```diff
@@ -89,36 +87,71 @@ class Timer {
 
 class DOM_Manager {
     constructor() {
-        this.cardsCountVar = document.getElementsByClassName("card").length;
         this.clickListenerBinded = this.clickListener.bind(this);
-        this.imgs = [];
-        for(let i=0;i<this.cardsCountVar;i++){
-            this.imgs.push(document.getElementById("img"+i));
-        }
+        this.submitOnClickBinded = this.submitOnClick.bind(this);
+        this.game=new Game();
+        document.getElementById("Easy").addEventListener("change", this.difficultOnChange);
+        document.getElementById("Medium").addEventListener("change", this.difficultOnChange);
+        document.getElementById("Hard").addEventListener("change", this.difficultOnChange);
+        document.getElementById("submit").addEventListener("click", this.submitOnClickBinded);
+    }
+    difficultOnChange() {
+        document.getElementById("submit").disabled = false;
     }
-    setListeners(fn) {
-        this.callBackFn = fn
-        for(let img of this.imgs) {
-            img.addEventListener('click', this.clickListenerBinded);
+    submitOnClick() {
+        const selectedDiff = document.querySelectorAll('input[name="difficulty"]:checked');
+        this.cardsCount = parseInt(selectedDiff[0].value);
+        document.getElementById("modalContainer").classList.add('hidden');
+        const cards = this.createCards();
+        this.game.start(cards);
+    }
+    createCards() {
+        const fragment=document.createDocumentFragment();
+        const cards = [];
+        for(let i = 0;i<this.cardsCount;i++) {
+            const card = new Card(i, this.cardsCount, this.clickListenerBinded);
+            cards.push(card);
+            fragment.appendChild(card.img);
         }
+        const mainContainer = document.getElementById("cardsContainer");
+        mainContainer.appendChild(fragment);
+        return cards;
     }
     clickListener(event) {
         const idClicked = parseInt(event.target.id.slice(3));
-        this.callBackFn(idClicked);
+        this.game.imageOnClickBinded(idClicked);
     }
-    get cardsCount() {
-        return this.cardsCountVar;
+}
+
+class Card {
+    constructor(number,count,clickListener) {
+        const img = document.createElement("img");
+        img.src="img/js-badge.svg";
+        img.alt=number;
+        img.id=`img${number}`;
+        img.order=Math.random() * 100;
+        img.classList.add(`cards${count}`, "card");
+        img.addEventListener("click", clickListener);
+        this.clickListener=clickListener;
+        this.img=img;
+        this.number=number;
+        this.opened=false;
+        this.visible=true;
     }
-    showFace(cardNumber, srcFileNumber) {
-        this.imgs[cardNumber].src = `img/${srcFileNumber}.svg`;
+    showFace(){
+        this.img.src=`img/${this.fileName}.svg`;

```

Comment:
I wonder what is inside `this.fileName`? And where it comes from? it seems like it's used only in this line